From dda8fc7e4312acc0353c320174a6c7a009b4c57f Mon Sep 17 00:00:00 2001
From: CrystalP <CrystalP@xbmc.org>
Date: Mon, 26 Dec 2011 14:17:03 -0500
Subject: [PATCH] fix crypted dvd playback

with recent changes to playdisc, CDVDInputStreamNavigator::Open now
receives a full path, which must be truncated more completely or libdvdcss
won't be able to decrypt.
---
 .../DVDInputStreams/DVDInputStreamNavigator.cpp    |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/xbmc/cores/dvdplayer/DVDInputStreams/DVDInputStreamNavigator.cpp b/xbmc/cores/dvdplayer/DVDInputStreams/DVDInputStreamNavigator.cpp
index d4cc77b..b9f42ca 100644
--- a/xbmc/cores/dvdplayer/DVDInputStreams/DVDInputStreamNavigator.cpp
+++ b/xbmc/cores/dvdplayer/DVDInputStreams/DVDInputStreamNavigator.cpp
@@ -80,6 +80,9 @@ bool CDVDInputStreamNavigator::Open(const char* strFile, const std::string& cont
   {
     strDVDFile[strlen(strDVDFile) - 13] = '\0';
   }
+  if (strncasecmp(strDVDFile + strlen(strDVDFile) - 8, "VIDEO_TS", 8) == 0)
+    strDVDFile[strlen(strDVDFile) - 9] = '\0';
+
 #if defined(__APPLE__) && !defined(__arm__)
   // if physical DVDs, libdvdnav wants "/dev/rdiskN" device name for OSX,
   // strDVDFile will get realloc'ed and replaced IF this is a physical DVD.
-- 
1.7.5.4

